File: OOP-and-Functional-Programming_Samples/OOP/Simulated_Assembly_instructions_and_Samples/java/Swap.java
public class Swap extends Instruccion{

    public void ejecutar(Programa programa){

        int c = programa.getPila().pop();
        int h = programa.getPila().pop();
        programa.getPila().push(c);
        programa.getPila().push(h);

        programa.inc();
    }

    public String mostrar(){
        return "swap";
    }
}

File: OOP-and-Functional-Programming_Samples/OOP/Simulated_Assembly_instructions_and_Samples/c++/swap.cc
#include "swap.h"
#include "Programa.h"
#include <stack>
using namespace std;

void Swap::ejecutar(Programa* programa){
    stack<int>& pila = programa->getPila();

    int c = pila.top();
    pila.pop();
    int h = pila.top();
    pila.pop();

    pila.push(c);
    pila.push(h);
    programa->inc();
}

string Swap::mostrar(){
    return "swap";
}
